ACP agent timeout after 120s of inactivity

## Problem

The ACP (Agent Control Protocol) agent becomes unresponsive and times out after 120 seconds of inactivity during prompt processing.

## Error Log

```
[WARN] [ACP] Prompt timed out after 120s of inactivity — agent unresponsive
[ERROR] [ACP] Prompt error: Error { code: -32603: Internal error, message: "Internal error", data: Some(String("Agent unresponsive — session will restart automatically")) }
[INFO] [ACP] Session force-stopped — exiting worker to kill agent process
```

## Timeline

1. Agent processes normally (tool calls, message chunks)
2. Agent goes silent for 120+ seconds
3. Timeout triggers, session force-killed
4. Session removed from map

## Likely Causes

1. **Agent hung internally** - Processing without emitting events
2. **IPC/stdio communication broke** - Agent running but not communicating
3. **Agent process crashed silently** - Killed by OS, undetected
4. **Long-running tool without progress** - Tool takes >120s without reporting

## Impact

- User's work is interrupted
- Session needs manual restart
- Poor UX when agents are doing legitimate long-running tasks

## Proposed Solutions

1. Increase timeout threshold for complex operations
2. Implement better progress reporting for long-running tools
3. Add heartbeat mechanism to distinguish between hung vs. busy agents
4. Provide user control over timeout settings

## Files Involved

- `src-tauri/src/acp.rs` - Timeout logic
- `src-tauri/src/acp/worker.rs` - Session lifecycle management

--- a/src-tauri/src/acp.rs
+++ b/src-tauri/src/acp.rs
@@ -194,9 +194,10 @@ fn auth_error_message(agent_type: AgentType) -> String {
     }
 }
 
-/// Maximum time to wait for an agent prompt to produce any response (2 minutes).
+/// Maximum time to wait for an agent prompt to produce any response (5 minutes).
 /// If the agent process is hung or the API is unreachable, this prevents infinite waits.
-const PROMPT_TIMEOUT_SECS: u64 = 120;
+/// Increased from 120s to allow complex operations like deep reasoning, refactoring, and multi-step tool execution.
+const PROMPT_TIMEOUT_SECS: u64 = 300;
 
 /// Debounce window: suppress duplicate login launches within this many seconds.
 const LOGIN_DEBOUNCE_SECS: i64 = 15;

